dot/network/service_test.go

```diff
@@ -72,17 +101,42 @@ func createTestService(t *testing.T, cfg *Config) (srvc *Service) {
 	}
 
 	if cfg.BlockState == nil {
-		cfg.BlockState = NewMockBlockState(nil)
+		parentHash := common.MustHexToHash("0x4545454545454545454545454545454545454545454545454545454545454545")
+		stateRoot := common.MustHexToHash("0xb3266de137d20a5d0ff3a6401eb57127525fd9b2693701f0bf5a8a853fa3ebe0")
+		extrinsicsRoot := common.MustHexToHash("0x03170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c111314")
+
+		header := &types.Header{
+			ParentHash:     parentHash,
+			Number:         big.NewInt(1),
+			StateRoot:      stateRoot,
+			ExtrinsicsRoot: extrinsicsRoot,
+			Digest:         types.NewDigest(),
+		}
+
+		blockstate := NewMockBlockState(ctrl)
+
+		blockstate.EXPECT().BestBlockHeader().Return(header, nil).AnyTimes()
+		blockstate.EXPECT().GetHighestFinalisedHeader().Return(header, nil).AnyTimes()
+		blockstate.EXPECT().GenesisHash().Return(common.NewHash([]byte{})).AnyTimes()
+		blockstate.EXPECT().BestBlockNumber().Return(big.NewInt(1), nil).AnyTimes()
+
+		blockstate.EXPECT().HasBlockBody(
+			gomock.AssignableToTypeOf(common.Hash([32]byte{}))).Return(false, nil).AnyTimes()
+		blockstate.EXPECT().GetHashByNumber(gomock.Any()).Return(common.Hash{}, nil).AnyTimes()
```

Comment:
Try to expect a concrete type if possible. Although maybe this is used everywhere and it's too complicated to do / out of scope, I'm not sure. Same goes for `AnyTimes()`, it would be better to expect a precise number of times.